libpromises/evalfunction.c

```diff
@@ -8136,6 +8142,41 @@ void ModuleProtocol(EvalContext *ctx, char *command, const char *line, int print
         }
         break;
 
+    case '&':
+        // TODO: the variable name is limited to 256 to accommodate the
+        // context name once it's in the vartable.  Maybe this can be relaxed.
+        sscanf(line + 1, "%256[^=]=%4095[^\n]", name, content);
+
+        if (CheckID(name))
+        {
+
+            if (FileCanOpen(content, "r"))
+            {
+                int size_max = IntFromString("inf");
+                const char *requested_mode = NULL;
+
+                requested_mode =  FileType(content);
+                Log(LOG_LEVEL_DEBUG, "Module protocol parsing %s file '%s'", requested_mode, content);
+
+                JsonElement *json = JsonReadDataFile("module file protocol", content, requested_mode, size_max);
+                if (json != NULL)
+                {
+                    Buffer *tagbuf = StringSetToBuffer(tags, ',');
+                    VarRef *ref = VarRefParseFromScope(name, context);
+
+                    EvalContextVariablePut(ctx, ref, json, CF_DATA_TYPE_CONTAINER, BufferData(tagbuf));
+                    VarRefDestroy(ref);
+                    BufferDestroy(tagbuf);
+                    JsonDestroy(json);
+                }
```

Comment:
You are not doing anything in case `JsonReadDataFile()` fails. It has internal error logging, so it might be okay, but I would add an `else` here with a comment.

```
else
{
    // reading / parsing failed, error message printed by JsonReadDataFile,
    // variable will not be created
}
```